src/net_processing.h

```diff
@@ -46,6 +46,8 @@ struct CNodeStateStats {
  * Memory is owned by shared pointers and this object is destructed when
  * the refcount drops to zero.
  *
+ * Mutexes inside this struct must not be held when locking m_peer_mutex.
```

Comment:
717a374e74b64b7b90bc1b2995e8900212bd0bfe

nit to avoid the inversion and give a rationale:

"m_peer_mutex should be locked before locking a mutex inside this struct to avoid lock order violations"

This will also weaken the requirement, because for some mutexes it could make sense to not require m_peer_mutex beforehand.

Reply:
That isn't precise enough. It's fine to lock m_peer_mutex, take a shared_ptr to one of the Peer objects, release m_peer_mutex and later take one of the inner locks. The only rule is that we don't want to take m_peer_mutex while already holding one of the inner locks.

This is the same as cs_vNodes and the inner locks in CNode.

Reply:
Ah right. I didn't parse this correctly. Thanks for the explanation.